pyproject.toml

```diff
@@ -1,32 +1,107 @@
 ]
-build-backend = "jupyter_packaging.build_api"
+build-backend = "hatchling.build"
 
-[tool.jupyter-packaging.options]
-skip-if-exists = [
-    "jupyterlab_latex/labextension/static/style.js"
+[project]
+name = "jupyterlab-latex"
+description = "JupyterLab extension for running LaTeX"
+readme = "README.md"
+requires-python = ">=3.8"
+authors = [
+    { name = "Jupyter Development Team", email = "jupyter@googlegroups.com" },
+]
+keywords = [
+    "Jupyter",
+    "JupyterLab",
+    "LaTeX",
+]
+classifiers = [
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "jupyter_server>=2,<3",
+    "jupyterlab>=4,<5",
+]
+version = "4.0.0"
+
+[project.license]
+file = "LICENSE"
+
+[project.urls]
+Homepage = "https://github.com/jupyterlab/jupyterlab-latex"
+
+[tool.hatch.build.targets.wheel.shared-data]
+"jupyterlab_latex/labextension/static" = "share/jupyter/labextensions/@jupyterlab/latex/static"
+"install.json" = "share/jupyter/labextensions/@jupyterlab/latex/install.json"
+"jupyterlab_latex/labextension/build_log.json" = "share/jupyter/labextensions/@jupyterlab/latex/build_log.json"
+"jupyterlab_latex/labextension/package.json" = "share/jupyter/labextensions/@jupyterlab/latex/package.json"
+"jupyterlab_latex/labextension/schemas/@jupyterlab/latex" = "share/jupyter/labextensions/@jupyterlab/latex/schemas/@jupyterlab/latex"
+"jupyter-config/server-config" = "etc/jupyter/jupyter_server_config.d"
+"jupyter-config/nb-config" = "etc/jupyter/jupyter_notebook_config.d"
+
+[tool.hatch.build.targets.sdist]
+exclude = [
+    ".github",
 ]
+
+[tool.hatch.build.hooks.jupyter-builder]
+dependencies = [
+    "hatch-jupyter-builder>=0.8.3",
+]
+build-function = "hatch_jupyter_builder.npm_builder"
 ensured-targets = [
     "jupyterlab_latex/labextension/package.json",
-    "jupyterlab_latex/labextension/static/style.js", 
+    "jupyterlab_latex/labextension/static/style.js",
+]
+skip-if-exists = [
+    "jupyterlab_latex/labextension/static/style.js",
 ]
 
-[tool.jupyter-packaging.builder]
-factory = "jupyter_packaging.npm_builder"
+[tool.hatch.build.hooks.jupyter-builder.editable-build-kwargs]
+build_dir = "jupyterlab_latex/labextension"
+source_dir = "src"
+build_cmd = "install:extension"
+npm = [
+    "jlpm",
+]
 
-[tool.jupyter-packaging.build-args]
+[tool.hatch.build.hooks.jupyter-builder.build-kwargs]
 build_cmd = "build:prod"
 npm = [
-    "jlpm"
+    "jlpm",
 ]
 
-[tool.check-manifest]
-ignore = [
-    ".*",
-    "jupyterlab_latex/labextension/**",
-    "package-lock.json",
-    "yarn.lock"
+[tool.tbump]
```

Comment:
If you switch to the new extension template, this will switch to using hatch directly aligning NPM and Python package version using: https://github.com/agoose77/hatch-nodejs-version